app/src/main/java/org/odk/odknotifications/DatabaseCommunicator/ServerDatabaseCommunicator.java

```diff
@@ -0,0 +1,261 @@
+        }
+
+        userDbInterface.insertRowWithId(appName,dbHandle,RESPONSES_TABLE_ID,orderedColumns,contentValues,response.getResponseID());
+
+    }
+
+    public static void addGroup(String groupId) throws ServicesAvailabilityException, ActionNotAuthorizedException {
+
+        OrderedColumns orderedColumns = userDbInterface.getUserDefinedColumns(appName,dbHandle,USER_TABLE_ID);
+
+
+        UserTable userTable = userDbInterface.simpleQuery(appName, dbHandle, USER_TABLE_ID, orderedColumns, null, null,
+                null,null,null,null,null,null);
+
+        int rowNo = userTable.getRowNumFromId(userId);
+        TypedRow typedRow = userTable.getRowAtIndex(rowNo);
+
+        String groups = typedRow.getStringValueByKey("GroupList");
+        String userName = typedRow.getStringValueByKey("UserName");
+
+        List<String> groupsArray = Arrays.asList(groups.split(","));
+
+        if(!groupsArray.contains(groupId)){
+            groups += (groupId + ",");
+        }
+
+        ContentValues contentValues = new ContentValues();
+
+        List<String>columnValues = Arrays.asList(userId,userName,groups);
+
+        for(int i=0;i<columnValues.size();i++){
+            contentValues.put(USERS_TABLE_COLUMNS_LIST.get(i) , columnValues.get(i));
+        }
+
+        userDbInterface.updateRowWithId(appName,dbHandle,USER_TABLE_ID,orderedColumns,contentValues,userId);
+    }
+
+    private static boolean isUserPresent(String userId) throws ServicesAvailabilityException {
+
+        OrderedColumns orderedColumns = userDbInterface.getUserDefinedColumns(appName,dbHandle,USER_TABLE_ID);
+
+        UserTable userTable = userDbInterface.simpleQuery(appName, dbHandle, USER_TABLE_ID, orderedColumns, null, null,
+                null,null,null,null,null,null);
+
+        boolean isUserPresent = false;
+
+        for(int i=0; i<userTable.getNumberOfRows(); i++){
+            if(userTable.getRowAtIndex(i).getStringValueByKey("UserId").equals(userId)){
+                isUserPresent = true;
+            }
+        }
+        return isUserPresent;
+    }
+
+    private static void addUser (String userId) throws ServicesAvailabilityException, JSONException, ActionNotAuthorizedException {
+
+        String roles_array_string = userDbInterface.getRolesList(appName);
+
+        JSONArray temp = new JSONArray();
+
+        if(roles_array_string!=null) {
+            temp = new JSONArray(roles_array_string);
+        }
+
+        StringBuilder groups = new StringBuilder();
+        for(int i=0; i<temp.length();i++){
+            if(temp.getString(i).startsWith("GROUP_")|| temp.getString(i).startsWith("ROLE_")) {
+                groups.append(temp.get(i)).append(",");
+            }
+        }
+
+        String userName = getUserName(userId);
+
+        List<String>columnValues = Arrays.asList(userId,userName,groups.toString());
+
+        ContentValues contentValues = new ContentValues();
+
+        for(int i=0;i<columnValues.size();i++){
```

Comment:
Try using for each loop

Reply:
I changed for loop to for-each loop wherever it is possible. But changing to a for-each loop, in the above-mentioned case is not possible.

Reply:
Got it, Thanks